spec/models/product_drive_spec.rb

```diff
@@ -42,6 +42,12 @@
     expect(result.count(0)).to eq(product_drive.organization.items.count - 1)
   end
 
+  it "calculates and returns all donated organization item quantities by category and date" do
+    create(:donation, :with_items, item_quantity: 2, product_drive: product_drive, issued_at: '26-01-2023')
+    result = product_drive.donation_quantity_by_date(Time.zone.parse('23/01/2023')..Time.zone.parse('26/01/2023'), 1)
+    expect(result).to eq product_drive.organization.items.where(item_category_id: 1).count
```

Comment:
This is a bit too finicky IMO. I'd actually create an item with the desired item category and verify that it comes back in the query. Otherwise we could just not have any data in there and it comes back empty... but we wouldn't know if it was empty because there was no data, or because there's an error in our query. In general for things like this you'll want at least one result that _should_ be included and one that _shouldn't_, and verify that only the one you want comes back.

Reply:
Good feedback, I've created two items with unique category ids and have the test verifying the one we want comes back